draft-ietf-taps-interface.md

```diff
@@ -2951,6 +2965,44 @@ and for contributing text, e.g., on multicast.
 
 --- back
 
+# Implementation Mapping {#implmapping}
+
+The way the concepts from this abstract interface map into concrete APIs in a
+given language on a given platform largely depends on the features and norms of
+the language and the platform. Actions could be implemented as functions or
+method calls, for instance, and Events could be implemented via event queues,
+handler functions or classes, communicating sequential processes, or other
+asynchronous calling conventions.
+
+## Types
+
+The basic types mentioned in {{notation}} typically have natural
+correspondences in practical programming languages, perhaps constrained by
+implementation-specific limitations. For example:
+
+- An Integer can typically be represented in C by an `int` or `long`, subject
+  to the underlying platform's ranges for each. To accommodate special values,
+  a C function that returns a non-negative `int` on success may return -1 on
+  failure. In Python, such a function might return `None` or raise an
+  exception.
+- In C, a Tuple may be represented as a `struct` with one member for each of
+  the value types in the ordered grouping. In Python, by contrast, a Tuple may
+  be represented natively as a `tuple`, a sequence of dynamically-typed
+  elements.
+- A Collection may be represented as a `std::set` in C++ or as a `set` in
+  Python. In C, it may be represented as an array or as a higher-level data
+  structure with appropriate accessors defined.
```

Comment:
This might be too implementation specific for the API draft and would maybe better fit into the implementation draft.

Reply:
I'm not sure... it's about the interface, and explains the abstract notation - it can help the reader mentally translate the text into their own language of choice.

For instance, Kyle himself probably didn't like the use of "-1" as an error because that's not his programming mindset...  and he probably thought that such a section would have helped him, as a reader. I think that applies to others in the same way - so I believe that this fits here.

Reply:
The purpose of this PR is to make the abstract API actually abstract. -1 is a C-ism, and C is about as inabstract as languages get. :-)

FWIW, I absolutely think the guidance in this appendix is appropriate for the interface draft because it's describing via example appropriate patterns for translating an abstract interface into an API in a real language, whereas the implementation draft is about how to implement the functionality that TAPS provides in a language-agnostic way.

Reply:
So this is resolved; sorry for mentioning "-1", I realize it was a bad example  :)